### ISSUE TITLE
Error setting app param Internal=Yes

### ISSUE THREAD

**Issue by apeace** [convox/rack#2002]
I came across an odd error, which looks like this:

<img src="http://i.imgur.com/xrp4V35.png" />

`CloudFormation cannot update a stack when a custom-named resource requires replacing. Rename ... and update the stack again.`

I started with an internal ELB like this:

```
  phpmyadmin:
    image: phpmyadmin/phpmyadmin:latest
    ports:
      - "80"
```

My goal was to change this to HTTPS-only. But since I can't change the port `phpmyadmin` will listen on, I updated the port mapping to `443:80`. This of course changed my ELB from internal to external.

Then I ran `convox apps params set Internal=Yes`, and observed the above CloudFormation error. My expectation was that this command would set the ELB to internal, yet maintain the `443:80` port mapping.

I have been unable to find a workaround for this, and so have had to change the mapping back to `80` for now.

**Comment by soulshake** [comment]
_Disclaimer: I'm checking with my colleagues to make sure the info I have on this issue is up to date, so my response below is pending confirmation._

The error `CloudFormation cannot update a stack when a custom-named resource requires replacing. Rename ... and update the stack again.` arises because CloudFormation will not replace custom-named resources. The custom-named resource here is the ELB. To change your app from internal to public-facing, the internal ELB needs to be deleted and replaced with an Internet ELB.

This shouldn't be happening on new apps since #341, so the best way to handle this would be to create a new app.

If that isn't possible, one workaround we've found is to set `Internal=No` and scale down the ELB to 0 at the same time. For example, if you have a process called `web`, the command would look like this:

`$ convox app params set Internal=No WebFormation=--1,0.256`

Once the update completes you should be able to scale the web count back up.

If you have more than one process in your app with a load balancer you'll need to scale down all of them by setting the first value in each `<processname>Formation` parameter to `-1`.

See also: https://github.com/convox/rack/issues/354

**Comment by apeace** [comment]
> This shouldn't be happening on new apps since #341

It looks like that was quite some time ago, and I just created this app last week. My rack version is `20170224012815`. I can try re-creating it if you think that will be the best remedy.

**Comment by soulshake** [comment]
For what it's worth, another user (@eladidan) ran into the same error today despite having a Rack version after the fix was released. It was apparently caused by the fact that the release they were trying to promote had been created before the Rack had been updated. By creating a new release, they were able to promote it successfully.

In your case, I think trying to recreate the app would be your best bet.

**Comment by apeace** [comment]
Here is what I did:

Updated my Convox CLI to `20170309072036`. My Rack version is `20170224012815`. I deleted my app, and recreated it.

Then I deployed the above `docker-compose.yml`, with just port 80. At this point setting `Internal=Yes` had no effect:

```
$ convox apps params set Internal=Yes
Updating parameters... ERROR: No updates are to be performed
```

So I changed my `docker-compose.yml` to use `443:80`, redeployed, and...

```
$ convox apps params set Internal=Yes
Updating parameters... OK
```

And I got the same error again:

```
CloudFormation cannot update a stack when a custom-named resource requires replacing.
```

It appears the bug is still present as of Rack `20170224012815`. Should I try updating the rack?
